agent/acs/handler/refresh_credentials_handler_test.go

```diff
@@ -233,6 +236,67 @@ func TestCredentialsMessageNotAckedWhenTaskNotFound(t *testing.T) {
 	cancel()
 }
 
+// TestCredentialsMessageNotAckedWhenDomainlessGMSACredentialsNotSet tests if credential messages
+// are not acked when the task arn in the message is not found in the task
+// engine
+func TestCredentialsMessageNotAckedWhenDomainlessGMSACredentialsError(t *testing.T) {
+	testCases := []struct {
```

Comment:
question(non-blocking): Since the test name suggests that we are testing a very specific scenario, do we need table-driven tests here?

Reply:
I have tested this manually over the last 2 days (have logs on the instance I can send)
You are right that we don't need table tests, but it was just in case for future extensibility. I can remove if needed.